### DISCUSSION TITLE
Disable command suggestions for individual Accounts

### DISCUSSION THREAD

**Discussion by Belameephus** [evennia/evennia#3182] [Evennia Questions] [answered]
I would like to have each Account be able to enable/disable command suggestions. Some of my Accounts/players prefer to not see any suggested commands when they enter a command that is not available.

Is it possible to turn off command suggestions individually for each Account, one Account sees suggestions while another does not?

if this is not possible, can command suggestions be turned off for ALL Accounts?

**Comment by aMiss-aWry** [comment, marked as answer]
https://www.evennia.com/docs/latest/Components/Commands.html#system-commands

If you want to alter what happens when various 'system commands' are triggered you can create a new command based on their syscmdkeys.key. So as stated in the above link there are 'system commands' triggered by no input (hitting enter with a blank command), a command that doesn't exist, a command with multiple matching results etc. You can change the result by creating a new command.

For instance, you could have something like this to **remove command suggestions for all accounts**:
```
from evennia import syscmdkeys

class NonexistentCommand(Command):
    key = syscmdkeys.CMD_NOMATCH

    def func(self):
        # Command not found
        self.caller.msg("Command not found.")
```

Then add `NonexistentCommand` to the base command set. If you want the same to happen when the command is blank, ambiguous, or so on, you can use the other keys described in the link above. This will override the [Evennia default behavior, found in commands/cmdhandler.py](https://github.com/evennia/evennia/blob/main/evennia/commands/cmdhandler.py#L737).

In order to create it as a setting for the user, you would need additional code to:

1. Add a command for users to toggle the setting on/off
2. Check whether the setting is toggled in the `NonexistentCommand`
3. Access and display suggestions if the setting is toggled on

**Reply by Belameephus** [reply]
I think I see how to accomplish a solution. Thanks much. Will comment once the code change in my game has been implemented and tested. Pretty sure I will have no trouble with the suggestions you provided.
